### PR TITLE
fix(support): #MO-145 fix escalation to redmine

### PR DESCRIPTION
**Pull request by vava56754**
## Describe your changes
Correction de l'escalade d'un ticket vers Redmine ou l'on rencontrait une erreur de NumberFormatException
## Checklist tests
- [x] Vérifier que l'escalade coté pivot fonctionne
## Issue ticket number and link
MO-145 : https://jira.support-ent.fr/browse/MO-145
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/net/atos/entng/support/controllers/TicketController.java [hunk 1]
```diff
@@ -732,12 +732,15 @@ private Handler<Either<String, JsonObject>> getEscalateTicketHandler(final HttpS
...
-                final JsonObject issue = escalationResponse.right().getValue().getJsonObject(Ticket.ISSUE, new JsonObject());
-                final String issueIdString = escalationService.getBugTrackerType().extractIdFromIssueString(issue);
-                final Number issueId = Integer.parseInt(issueIdString);
+                final JsonObject issue = escalationResponse.right().getValue();
+                final Number issueId = escalationService.getBugTrackerType().getIssueId(issue);
 
+                if(issueId != null){
+                    escalationService.getIssue(issueId, getIssueHandler(request, issueId, ticketId, user, attachmentMap, doResponse, issue));
+                } else {
+                    renderError(request, new JsonObject().put("error","Failed to parse integer"));
```
Comment (JinDouang) [inline src/main/java/net/atos/entng/support/controllers/TicketController.java:741]: une badRequest plutôt non ?

File: src/main/java/net/atos/entng/support/enums/BugTracker.java [hunk 2]
```diff
@@ -60,6 +67,19 @@ public BugTrackerSyncType getBugTrackerSyncType() {
...
 	},
 	PIVOT {
+        @Override
+        public Number getIssueId(JsonObject issue) {
+            Number issueId = 0;
+            String issueIdString = this.extractIdFromIssueString(issue.getJsonObject(Ticket.ISSUE, new JsonObject()));
+            try {
+                issueId = Integer.parseInt(issueIdString);
+            } catch (NumberFormatException e) {
+				e.printStackTrace();
```
Comment (JinDouang) [inline src/main/java/net/atos/entng/support/enums/BugTracker.java:77]: indentation espace..

File: src/main/java/net/atos/entng/support/enums/BugTracker.java [hunk 3]
```diff
@@ -60,6 +67,19 @@ public BugTrackerSyncType getBugTrackerSyncType() {
 		}
 	},
 	PIVOT {
+        @Override
+        public Number getIssueId(JsonObject issue) {
+            Number issueId = 0;
```
Comment (JinDouang) [inline src/main/java/net/atos/entng/support/enums/BugTracker.java:72]: Pourquoi tu l'assignes à 0 ?

File: src/main/java/net/atos/entng/support/controllers/TicketController.java [hunk 4]
```diff
@@ -732,12 +732,15 @@ private Handler<Either<String, JsonObject>> getEscalateTicketHandler(final HttpS
 
         return escalationResponse -> {
             if (escalationResponse.isRight()) {
-                final JsonObject issue = escalationResponse.right().getValue().getJsonObject(Ticket.ISSUE, new JsonObject());
-                final String issueIdString = escalationService.getBugTrackerType().extractIdFromIssueString(issue);
-                final Number issueId = Integer.parseInt(issueIdString);
+                final JsonObject issue = escalationResponse.right().getValue();
+                final Number issueId = escalationService.getBugTrackerType().getIssueId(issue);
 
+                if(issueId != null){
```
Comment (JinDouang) [inline src/main/java/net/atos/entng/support/controllers/TicketController.java:738]: indentation espace

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by flmartineau — APPROVED** [review]

**Review by JinDouang — APPROVED** [review]

**Pull request merged**